src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -473,83 +492,62 @@ private void UpdateValueInFile<T>(ConfigScope scope, string key, T value, bool a
+                //
+                using var stream = OpenFileStreamWithRetry(fileName, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.None);
+                using var reader = new StreamReader(stream);
+                string jsonString = reader.ReadToEnd();
 
-                JObject jsonObject = null;
-                using FileStream fs = OpenFileStreamWithRetry(fileName, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.None);
+                Dictionary<string, JsonElement> configScopeData = null;
+                if (!string.IsNullOrEmpty(jsonString))
+                {
+                    configScopeData = JsonSerializer.Deserialize<Dictionary<string, JsonElement>>(jsonString);
+                }
 
-                // UTF8, BOM detection, and bufferSize are the same as the basic stream constructor.
-                // The most important parameter here is the last one, which keeps underlying stream open after StreamReader is disposed
-                // so that it can be reused for the subsequent write operation.
-                using (StreamReader streamRdr = new StreamReader(fs, Encoding.UTF8, detectEncodingFromByteOrderMarks: true, bufferSize: 1024, leaveOpen: true))
-                using (JsonTextReader jsonReader = new JsonTextReader(streamRdr))
+                if (configScopeData is null)
                 {
-                    // Safely determines whether there is content to read from the file
-                    bool isReadSuccess = jsonReader.Read();
-                    if (isReadSuccess)
-                    {
-                        // Read the stream into a root JObject for manipulation
-                        jsonObject = serializer.Deserialize<JObject>(jsonReader);
-                        JProperty propertyToModify = jsonObject.Property(key);
+                    configScopeData = new Dictionary<string, JsonElement>();
+                }
 
-                        if (propertyToModify == null)
-                        {
-                            // The property doesn't exist, so add it
-                            if (addValue)
-                            {
-                                jsonObject.Add(new JProperty(key, value));
-                            }
-                            // else the property doesn't exist so there is nothing to remove
-                        }
-                        else
-                        {
-                            // The property exists
-                            if (addValue)
-                            {
-                                propertyToModify.Replace(new JProperty(key, value));
-                            }
-                            else
-                            {
-                                propertyToModify.Remove();
-                            }
-                        }
+                var v = JsonSerializer.Serialize(value);
```

Comment:
These conversions between `JsonElement` and strongly-typed objects here and in `ToObject<T>()` feel wrong — there must be a better way to do this